### ISSUE TITLE
Messages in journalist_templates/js-strings.html not working

### ISSUE THREAD

**Issue by redshiftzero** [freedomofpress/securedrop#2651]
# Description

Messages in `journalist_templates/js-strings.html` are not being displayed in the correct language.

# Steps to Reproduce

Look at the following areas on the journalist application.

<img width="872" alt="screen shot 2017-11-30 at 4 42 35 pm" src="https://user-images.githubusercontent.com/7832803/33462852-0472323a-d5ef-11e7-9705-081ff20bce92.png">

"Select All", "Select None":

<img width="992" alt="screen shot 2017-11-30 at 4 45 55 pm" src="https://user-images.githubusercontent.com/7832803/33462813-bb376b1c-d5ee-11e7-9294-c1473b8e739e.png">

# Expected Behavior

Strings appear in the selected locale

# Actual Behavior

Strings appear in English

**Comment by dachary** [comment]
This cannot be reproduced in a development environment. Trying with production.

**Comment by dachary** [comment]
I'm unable to reproduce the problem. The only differences I see that may have an impact are:

* Not upgrade from 0.4.4 to 0.5 but installed 0.5 fresh (although I don't see how it would make any difference given that the translations did not exist in 0.4.4)
* Packages are make with **make build-debs** and obtained from **https://packages.securedrop.club/** (because I have hope it could be used for the next release QA to not require manual intervention)

**Comment by dachary** [comment]
Here is an idea: the .js is cached from a visit to the 0.4.4 journalist page. It is not reloaded and still contains the un-gettextized strings from 0.4.4. If that's the cause, a Control-F5 will fix the problem.

**Comment by redshiftzero** [comment]
Good suggestion, but a Control-F5 did not fix the problem. After further investigation, it looks like `/var/www/securedrop/static/gen/journalist.js` _does not_ contain the updated js, but `/var/www/securedrop/static/js/journalist.js` _does_ contain the updated js, and the HTML is pointing to `/var/www/securedrop/static/gen/journalist.js`. So that points to an issue with `Flask-Assets`. 

Btw I found this issue in a 0.4.4->0.5 upgrade in `0.5-rc2` prod VMs. Digging...

**Comment by dachary** [comment]
That's going to be an interesting one I'm sure.

**Comment by redshiftzero** [comment]
I removed the existing file in `/var/www/securedrop/static/gen/journalist.js` as a test, which resolved the issue and the strings appear in the localized file as expected.

To determine if Flask-Assets was building the static assets properly after a change, I then changed the underlying `/var/www/securedrop/static/js/journalist.js` and Flask-Assets updated the file as expected.

Going to pause on this one until someone else reproduces this as an issue, else in my next upgrade test I will see if this happens again and investigate further at that time.

**Comment by emkll** [comment]
Cannot reproduce this bug in `0.5-rc-2`

**Comment by kushaldas** [comment]
I am wondering the following is also because of same issue. Updated from 0.4.4 to rc2 to rc3 in this instance.

![flash_english_in_de](https://user-images.githubusercontent.com/272303/33504712-813adab8-d70e-11e7-81bb-276da7ab1526.png)

**Comment by redshiftzero** [comment]
Ah that one @kushaldas was due to the WTForm localization issue in issue #2644, the fix was brought into the release branch (after `rc3` which is why it's still happening) in #2650, but this should be fixed when we cut `rc4` later today

**Comment by redshiftzero** [comment]
I was able to reproduce this via:

1. Fresh install of `0.4.4`
2. Load journalist interface in Tor Browser (thus ensuring the static assets are generated using the 0.4.4 js)
3. Upgrade to `0.5-rc3`
4. Sign into journalist interface

<img width="764" alt="screen shot 2017-12-01 at 3 49 36 pm" src="https://user-images.githubusercontent.com/7832803/33508548-d01c48ba-d6af-11e7-97db-efc61706bcfa.png">

**Comment by dachary** [comment]
Good sleuth work !

**Comment by redshiftzero** [comment]
Dug a bit more into why this is happening. [`Flask-Assets`](https://github.com/miracle2k/flask-assets/blob/master/src/flask_assets.py#L12) uses Python's `webassets` to rebuild static assets, and `webassets` will rebuild the static assets in `static/gen` only if one of the following conditions is true:

1. `static/gen` is empty OR
2. the timestamps of the source files in `static/js` are more recent than the timestamp on the existing files in `static/gen` ([see here](https://github.com/marcellarius/webassets/blob/d1a4881295cb8b1c777b1a0301005e816d729279/src/webassets/env.py#L597)). 

When I provisioned a prod VM on SecueDrop 0.4.4 today, the timestamps on the files in `static/js` were mid ~October. When I visited the journalist interface on 0.4.4, `Flask-Assets` generated `static/gen/journalist.js` with a timestamp of today - December 3rd (UTC):

<img width="734" alt="after-visiting-the-journalist-interface-still-0-4-4" src="https://user-images.githubusercontent.com/7832803/33521282-cf46d544-d782-11e7-886b-5d4bfc86fa87.png">

Then I upgraded the prod VMs to 0.5-rc3 and visited the journalist interface again. Since the timestamps on the `js` files was then mid-November, the static assets in `static/gen` were more recent than `static/js`:

<img width="693" alt="after-upgrading-to-05" src="https://user-images.githubusercontent.com/7832803/33521297-110d7c9e-d783-11e7-98ef-6e1058956f7f.png">

So, the static assets in `static/gen` were not regenerated. In order to prevent this situation from arising, we should make sure when we ship updates that we force a rebuild of static assets. The simplest way to do this would be to delete the current contents of `static/gen` (e.g. in `postinst`) so that the first visit to either interface after update triggers a rebuild.
